### DISCUSSION TITLE
Move to built-in standard Cmake YACC/LEXX?

### DISCUSSION THREAD

**Discussion by wrosecrans** [AcademySoftwareFoundation/OpenRV#70] [Ideas]
This is more of a style choice question than a bug report.

OpenRV is using custom macros LEXX_IT() and YACC_IT() in macros/rv_yacc.cmake and macros/rv_lex.cmake for the GTO/Mu grammars.

CMake comes with some convenience functions for FLEX and Bison  https://cmake.org/cmake/help/latest/module/FindFLEX.html and OpenRV already requires a new enough CMake to depend on it.

Does the project want to eliminate custom macros and move to the off the shelf stuff?  Requires a little elbow grease, but I think it will ultimately make it easier to deal with build issues because problems with it will be more googleable and allow deleting a bit of code.  Relatedly and probably not worth asking in a separate issue, OpenRV uses custom types like "int32" would there be interest in moving that stuff to stdint types like uint32_t?

**Comment by rogernelson** [comment]
The CMake system is very new (it was done as a part of the work to open source), so there may be many places where we are not yet fully leveraging it.  Any pull requests to do so would definitely be welcome!

Same for using standard types (I assume you mean int32 -> int32_t (not uint32_t).  Any PRs that replace non-standard types with standards are also welcome.
